### ISSUE TITLE
Saving World While Uploading With SDK Results in Unpredictable Behaviour

### ISSUE THREAD

**Issue by Enverex** [Frooxius/NeosPublic#463]
I've uploaded a few large worlds lately (and some that aren't that large, but my internet connection makes it seem that way) and I've run into an issue when trying to save, though the behaviour isn't 100% reliable in terms of recreation.

There's no way to know when anything has completed currently. Right now, you initiate the "setup" by dragging the world object onto the component, Unity locks up for a while and then things carry on, but there's no way to really know what is going on here. That in itself isn't an issue but ties into the following:

If a world is still uploading content, it appears to become unsavable, or perhaps queues the save after everything else. On two occasions (while the world contents were still uploading I assume) I have tried to save a world, and on both occasions Unity froze up completely for a few seconds, then resumed, then showed a loading spinner around the world orb. After a while, the orb would remain and no world would appear in your inventory (and the world wouldn't appear in "Your Worlds" either).

Waiting a while then trying again in one of the worlds resulted in the world orb disappearing and it being transferred to my inventory. I then closed the SDK, opened the normal client (which claimed to be In Sync) but the world orb wasn't in my inventory and the world wasn't showing in the world browser. I loaded the SDK again and went into play mode (which still claimed to be in sync) and just left it for a few minutes. I then closed it again, reopened the normal client and the orb and world browser entry were now present.

So in short; world saving is unreliable while the content still seems to be uploading and certain actions seem to end up queued somewhere but eventually go through.

**Comment by TehTurk** [comment]
This kinda makes sense, I know Neos in game is optimized for saving but the SDKs upload procedures might not queue properly in the cloud. Even when you save stuff in Neos it takes a second to show up in your cloud because it needs to Sync.
